packages/components/modules/navigations/Header/AccountMenu/AccountPopover/__storybook__/stories.tsx

```diff
@@ -1,37 +1,87 @@
+import { CURRENT_PROFILE_STORAGE_KEY } from '../../../../../profiles/context/CurrentProfileProvider/constants'
+import { ProfilesListQuery } from '../../../../../profiles/graphql/queries/ProfilesList'
 import { AccountPopoverProps } from '../types'
+import { mockResolvers } from './mockResolvers'
+
+UserApi.getUser = async <TUser extends Partial<User>>() =>
+  ({
+    id: 1,
+    firstName: 'Mocked',
+    lastName: 'User',
+    email: 'mocked.user@example.com',
+  }) as TUser
 
 export default {
   title: '@baseapp-frontend | components/Navigation/AccountPopover',
   component: AccountPopover,
   args: {
-    menuItems: [
-      { label: 'Profile', onClick: () => console.log('Profile clicked') },
-      { label: 'Settings', onClick: () => console.log('Settings clicked') },
-    ],
+    MenuItemsProps: {
+      menuItems: [
+        { label: 'Profile', onClick: () => console.log('Profile clicked') }, // eslint-disable-line no-console
+        { label: 'Settings', onClick: () => console.log('Settings clicked') }, // eslint-disable-line no-console
+      ],
+    },
   },
   argTypes: {
-    menuItems: { control: 'object' },
-    logoutButtonLabel: { control: 'text' },
-    hideLogoutButton: { control: 'boolean' },
+    popoverStyles: { control: 'object', table: { type: { summary: 'CSSProperties' } } },
+    CurrentUser: { control: false, table: { type: { summary: 'FC' } } },
+    CurrentProfile: { control: false, table: { type: { summary: 'FC' } } },
+    MenuItems: { control: false, table: { type: { summary: 'FC<MenuItemsProps>' } } },
+    MenuItemsProps: { control: 'object', table: { type: { summary: 'MenuItemsProps' } } },
+    SwitchProfileMenu: {
+      control: false,
+      table: { type: { summary: 'FC<SwitchProfileMenuProps>' } },
+    },
+    SwitchProfileMenuProps: {
+      control: 'object',
+      table: { type: { summary: 'SwitchProfileMenuProps' } },
+    },
+    ProfilesList: { control: false, table: { type: { summary: 'FC<ProfilesListProps>' } } },
+    ProfilesListProps: { control: 'object', table: { type: { summary: 'ProfilesListProps' } } },
+    AddProfileMenuItem: {
+      control: false,
+      table: { type: { summary: 'FC<AddProfileMenuItemProps>' } },
+    },
+    AddProfileMenuItemProps: {
+      control: 'object',
+      table: { type: { summary: 'AddProfileMenuItemProps' } },
+    },
+    LogoutItemProps: { control: 'object', table: { type: { summary: 'LogoutItemProps' } } },
   },
   tags: ['autodocs'],
-  decorators: [withTokenSetup],
+  decorators: [
+    withTokenSetup,
+    (Story, context) => {
+      localStorage.removeItem(CURRENT_PROFILE_STORAGE_KEY)
+
+      const { queueOperationResolver, queuePendingOperation } = context.parameters
+        .relayMockEnvironment as ReturnType<typeof createTestEnvironment>
+
+      queueOperationResolver(mockResolvers)
+      queuePendingOperation(ProfilesListQuery)
+
+      return (
+        <CurrentProfileProvider>
```

Comment:
I think we can remove that once we add it under `packages/components/.storybook/decorators/withProviders.tsx `